doc/modules/calibration.rst

```diff
@@ -111,30 +124,83 @@ predicted probabilities of the `k` estimators in the
 `calibrated_classifiers_` list. The output of :term:`predict` is the class
 that has the highest probability.
 
-The regressor that is used for calibration depends on the `method`
-parameter. `'sigmoid'` corresponds to a parametric approach based on Platt's
-logistic model [3]_, i.e. :math:`p(y_i = 1 | f_i)` is modeled as
-:math:`\sigma(A f_i + B)` where :math:`\sigma` is the logistic function, and
-:math:`A` and :math:`B` are real numbers to be determined when fitting the
-regressor via maximum likelihood. `'isotonic'` will instead fit a
-non-parametric isotonic regressor, which outputs a step-wise non-decreasing
-function (see :mod:`sklearn.isotonic`).
-
-An already fitted classifier can be calibrated by setting `cv="prefit"`. In
-this case, the data is only used to fit the regressor. It is up to the user
+Alternatively an already fitted classifier can be calibrated by setting
+`cv="prefit"`. In this case, the data is not split and all of it is used to
+fit the regressor. It is up to the user
 make sure that the data used for fitting the classifier is disjoint from the
 data used for fitting the regressor.
 
-:class:`CalibratedClassifierCV` can calibrate probabilities in a multiclass
-setting if the base estimator supports multiclass predictions. The classifier
-is calibrated first for each class separately in a one-vs-rest fashion [4]_.
-When predicting probabilities, the calibrated probabilities for each class
+:func:`sklearn.metrics.brier_score_loss` may be used to assess how
+well a classifier is calibrated. However, this metric should be used with care
+because a lower Brier score does not always mean a better calibrated model.
+This is because the Brier score metric is a combination of calibration loss
+and refinement loss. Calibration loss is defined as the mean squared deviation
+from empirical probabilities derived from the slope of ROC segments.
+Refinement loss can be defined as the expected optimal loss as measured by the
+area under the optimal cost curve. As refinement loss can change
+independently from calibration loss, a lower Brier score does not necessarily
+mean a better calibrated model.
+
+:class:`CalibratedClassifierCV` supports the use of two 'calibration'
+regressors: 'sigmoid' and 'isotonic'.
+
+Sigmoid
+^^^^^^^
+
+The sigmoid regressor is based on Platt's logistic model [3]_:
+
+.. math::
+       p(y_i = 1 | f_i) = \frac{1}{1 + \exp(A f_i + B)}
+
+where :math:`y_i` is the true label of sample :math:`i` and :math:`f_i`
+is the output of the un-calibrated classifier for sample :math:`i`. :math:`A`
+and :math:`B` are real numbers to be determined when fitting the regressor via
+maximum likelihood.
+
+The sigmoid method is biased in that it assumes the :ref:`calibration curve
+<calibration_curve>` of the un-calibrated model has a sigmoid shape and is
+symmetrical [1]_. It is thus most effective when the un-calibrated model is
```

Comment:
I don't want to delay merging further if you're sure about this but there are a few things that aren't clear for me here:

- why does sigmoid calibration assumes a sigmoid calibration curve?
- is this really discussed in [1]_, or rather in projecteuclid.org/download/pdfview_1/euclid.ejs/1513306867? 

In projecteuclid.org/download/pdfview_1/euclid.ejs/1513306867  it is said that

> the parametric assumption made by logistic calibration is exactly the right one if the scores output by a classifier are normally distributed within each class around class means s+ and s− with the same variance σ2

though I'm not sure yet how that relates to the comment made above

Reply:
The symmetry assumption is discussed in projecteuclid.org/download/pdfview_1/euclid.ejs/1513306867 but not in '`[1]`'. I can add it in.

>     * why does sigmoid calibration assumes a sigmoid calibration curve?

I am not clear on the maths but I think this is explained better in the original Platt paper: https://www.researchgate.net/publication/2594015_Probabilistic_Outputs_for_Support_Vector_Machines_and_Comparisons_to_Regularized_Likelihood_Methods
 section 2.1 (which I can reference instead)

Reply:
> I am not clear on the maths but I think this is explained better in the original Platt paper

I don't see where this paper says such a thing. What I read is:

- the sigmiod model is equivalent to assuming that the output of the SVM is proportional to the log odds of a positive example
- the class-conditional densities between the margins are apparently exponential. Bayes rules on 2 exponentials suggests using a parametric form of a sigmoid

I don't understand how these two are equivalent to "using a sigmoid calibration assumes that the calibration curve has a sigmoid shape"

Reply:
Hmm yes that is true. It was designed to calibrate the output of the SVM - which has a sigmoid shape (is this always true)?

Reply:
Maybe we can just say:

- using the sigmoid calibration method assumes that the calibration curve can fixed by applying a sigmoid function on the predictions. This assumption has been empirically justified in the case of support vector machines with common kernel functions on various benchmark datasets in section 2.1 of Platt 2000 [1] but does not necessarily hold in general.